week-06-intro-ruby-on-rails/day-03/module-01/readme.md

```diff
@@ -1 +1,187 @@
 # <img src="https://cloud.githubusercontent.com/assets/7833470/10899314/63829980-8188-11e5-8cdd-4ded5bcb6e36.png" height="60"> Ruby OOP
+
+| Objectives |
+| :-- |
+| Distinguish between objects and classes in Ruby. |
+| Define attributes and instance methods. |
+| Explain the difference between instance and class variables. |
+
+<img src="https://cloud.githubusercontent.com/assets/1329385/11281896/8aedc310-8eb1-11e5-8eff-a6a3a339e747.gif">
+
+> Even an Object is an Object.
+
+## Hashes
+
+Let's create a new Hash
+
+* Hashes are simple key/value stores.
+
+**Challenge:**
+
+How can I organize my data using key/value pairs in Ruby? Like so:
+
+```ruby
+# hash rocket notation
+# accessed like hash["key"] #=> value
+hash = {"name"=>"Napoleon", "fav_food"=>"steak", "skills"=>["archery", "combat", "egg farming"]}
+hash["name"] #=> "Napoleon"
+hash['skills'].first #=> "archery"
+test =  { 0 => "Zero", 1 => "One", 2 => "Two" }
+test[0] #=> "Zero"
+
+website = { 'url' => 'http://myblog.wordpress.com', 'author' => 'Melissa ', 'likes' => 3000 }
+website['url'] #=> 'http://myblog.wordpress.com'
+website['likes'] #=> 3000
+
+# symbol notation, use this more often
+# accessed like hash[:key] #=> value
+agent = {name: "James Bond", fav_food: "Vodka Martini",
+skills: ["hand to hand combat", "marksmanship", "espionage"]}
+
+agent[:name] #=> "James Bond"
+agent[:skills] #=> ["hand to hand combat", "marksmanship", "espionage"]
+```
+
+## Objects
+
+Everything in Ruby is an Object; however, we almost never use plain vanilla Objects because there are more sophisticated implementations of them such a `String`, `Array`, `Integer` and `Hash`.
+
+Different objects provide different types of functionality for storing data and performing behaviors. However, for more complex applications, classes (a certain kind of object) are used to model real-world interactions.
+
+**Challenge:**
+How can we prove that the Hash we just created inherited from `Basic Object`?
+
+*Hint:* use the `#is_a?` method
+
+### Class Inheritance Tree
+
+![Class inheritance](http://i.stack.imgur.com/rvcEi.png)
+
+##Classes
+
+**Challenge:**
```

Comment:
Can you make one of those drop-down arrows here with the solution to this "challenge". Just so they have the code example to refer back to in the notes.